### PR TITLE
feat: add flexibility for less masters/workers and memory per vm

### PR DESCRIPTION
**Pull request by sebastienbonami**

### CODE REVIEW

File: shared/params.json [hunk 1]
```diff
@@ -22,5 +22,16 @@
...
+
+    "etcd_memory": 512,
+    "coredns_memory": 512,
+    "k8_bastion_memory": 512,
+    "k8_lb_memory": 512,
+    "k8_masters_memory": 2048,
+    "k8_workers_memory": 1536,
+
+    "k8_masters_count": 1,
+    "k8_workers_count": 1
```
Comment (Magnitus-) [inline shared/params.json]: I think realistically, if you want to troubleshoot with an HA setup to have higher fidelity with what an installation in deployed environments is like, you can get away with 1 worker, but you kind of have to put 3 masters though.

Also, I'd make the default configs something that is sensible on a machine with more meat.

Though you could definitely create an alternate config file for a 16GB ram machine.

File: README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ External dependencies are linked via git submodules with an **ssh** link for eas
 
 # Requirements
 
-A Linux machine with modern multi-core cpus and at least 64GB RAM is recommend.
+A Linux machine with modern multi-core cpus and at least 16GB RAM is required.
```
Comment (Magnitus-) [inline README.md]: We can add 16GB RAM required, but I'd add 32GB RAM or more as heavily recommended for a good experience.

I think people running this on a 16GB RAM machine are bound to have an okish experience with this at best, especially if they have something else going on at the same time. I'd rather be upfront about it.

File: README.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ In all cases, you will want to perform the following steps:
 
 This translates into the following steps:
 
-1. Inspect the **shared/params.json** file and change it if needed
+1. Inspect the **shared/params.json** file and change it if needed (if you have more than 16GB RAM, feel free to increase `memory` and `count` values)
```
Comment (Magnitus-) [inline README.md]: I'd go the other way about it with a default appropriate for a better machine and a file with alternate minimum values they can swap if their specs are less.

**Comment by Magnitus-** [comment]
Nice DRY :).

We'll still have to be more verbose on deployed environments because terraform didn't allow dynamically setting the provider value last I checked (which the libvirt provider uses in multi-machine setups), but definitely appropriate here.

File: shared/params.json (hunk 1 above)
Comment (sebastienbonami) [reply shared/params.json]: Done.

File: README.md (hunk 2 above)
Comment (sebastienbonami) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (sebastienbonami) [reply README.md]: Done.

**Pull request merged**